docs/Encryption.md

```diff
@@ -0,0 +1,76 @@
+S3Proxy
+
+# Encryption 
+
+## Motivation 
+The motivation behind this implementation is to provide a fully transparent and secure encryption to the s3 client while having the ability to write into different clouds.
+
+## Cipher mode
+The chosen cipher is ```AES/CFB/NoPadding``` because it provides the ability to read from an offset like in the middle of a ```Blob```.
+While reading from an offset the decryption process needs to consider the previous 16 bytes of the AES block.
```

Comment:
Store the cipher type in user metadata?  This allows having mixed encryption types in case the data format ever changes.

Reply:
There is a version in the padding that can be used if it is needed in the future.